clever deploy don't support app with colon in name

I started to create applications and add-ons with colon in the name.
But it's impossible to use `clever deploy` on the applications.

I don't know if it's a bug of the `clever deploy` command or not. 
But it seems your remote git server uses internally the app name 😅 

### Sample

```shell
$ clever create "app:with:colon"
$ clever deploy
[ERROR] "app:with:colon" would be an invalid git reference. (Hint: a valid alternative would be "app-with-colon".)
```

### Workaround

Stop using colon `:` in your app name.

--- a/src/lib/slugify.js
+++ b/src/lib/slugify.js
@@ -1,0 +1,10 @@
+import slugifyRaw from 'slugify';
+
+/**
+ * Converts a string to a slug using strict mode
+ * @param {string} string - The string to be converted to a slug
+ * @returns {string} The slugified string
+ */
+export function slugify (string) {
+  return slugifyRaw(string, { strict: true });
+}

--- a/src/models/app_configuration.js
+++ b/src/models/app_configuration.js
@@ -2,7 +2,7 @@ import { promises as fs } from 'node:fs';
 import path from 'node:path';
 
 import _ from 'lodash';
-import slugify from 'slugify';
+import { slugify } from '../lib/slugify.js';
 
 import { Logger } from '../logger.js';
 import * as User from './user.js';

--- a/src/models/git.js
+++ b/src/models/git.js
@@ -5,7 +5,7 @@ import _ from 'lodash';
 import git from 'isomorphic-git';
 import * as http from './isomorphic-http-with-agent.js';
 import cliparse from 'cliparse';
-import slugify from 'slugify';
+import { slugify } from '../lib/slugify.js';
 import { findPath } from './fs-utils.js';
 import { loadOAuthConf } from './configuration.js';
 

